src/odemis/gui/cont/microscope.py

```diff
@@ -325,6 +340,41 @@ def __init__(self, tab_data, tab_panel, btn_prefix, st_ctrl):
             if ch_pos.value["pressure"] == self._overview_pressure:
                 self._main_data.chamberState.value = CHAMBER_PUMPING
 
+        # disable optical and SEM buttons while there is a preparation process running
+        self._main_data.is_preparing.subscribe(self.on_preparation)
+
+    def _on_stage_move(self, pos):
+        """
+        Called when the stage moves (changes position)
+        pos (dict): new position
+        """
+        # Check if the position has really changed, as some stage tend to
+        # report "new" position even when no actual move has happened
+        if self._last_pos == pos:
+            return
+        self._last_pos = pos
+        self._stage_time = time.time()
+        self._remove_misaligned()
+        self.decide_status()
+
+    def _on_focus_move(self, pos):
+        """
+        Called when the focus moves (changes position)
+        pos (dict): new position
+        """
+        self._focus_time = time.time()
+        for s in self._tab_data.streams.value:
+            if model.hasVA(s, "calibrated"):
+                s.calibrated.value = False
```

Comment:
Why do you have to do this on the focus move but not on the stage move?
Preferably, the code for stage and focus move should be shared, and use only one variable to store the last changes.